File: crates/mct-lang-java/tests/fixtures/billing-app/Invoice.java
package billing;

public class Invoice {
    private double total;

    public double addItem(double price) {
        total += price;
        Logger.log("item added");
        return total;
    }

    public double addItem(double price, double taxRate) {
        double taxed = price * (1 + taxRate);
        total += taxed;
        Logger.log("item added with tax");
        return total;
    }
}

File: crates/mct-lang-cpp/tests/fixtures/billing-app/Invoice.cpp
#include "Invoice.h"
#include "Logger.h"

double Invoice::addItem(double price) {
    total += price;
    Logger::log("item added");
    return total;
}

double Invoice::addItem(double price, double taxRate) {
    double taxed = price * (1 + taxRate);
    total += taxed;
    Logger::log("item added with tax");
    return total;
}
